File: backend/src/main/java/de/navvis/pokerplanning/room/web/rest/RoomStatus.java
package de.navvis.pokerplanning.room.web.rest;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

import de.navvis.pokerplanning.room.web.domain.UserEstimate;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RoomStatus
{
	/**
	 * The list of all estimates for all users.
	 */
	private List<UserEstimate> estimates;

	/**
	 * Whether the voting phase is ongoing. If false, means results can be displayed.
	 */
	private boolean isVotingOngoing;

	/**
	 * The id of the room moderator.
	 */
	private UUID moderatorId;

	/**
	 * The id of the current user connected to the room.
	 */
	private UUID userId;
}

File: frontend/src/app/services/room/domain/RoomStatus.ts
export interface RoomStatus
{
	/**
	 * The list of all estimates for all users.
	 */
	estimates: UserEstimate[];

	/**
	 * Whether the voting phase is ongoing. If false, means results can be displayed.
	 */
	votingOngoing: boolean;

	/**
	 * The id of the room moderator.
	 */
	moderatorId: string;

	/**
	 * The id of the user connected to the room.
	 */
	userId: string;
}

export interface UserEstimate
{
	/**
	 * The name of the user.
	 */
	username: string;

	/**
	 * The unique id of the user.
	 */
	userId: string;

	/**
	 * Whether this user has provided an estimate.
	 */
	voted: boolean;

	/**
	 * The estimate this user has selected. Can be null if {@link #voted} is false.
	 */
	estimate: string;
}
